session/promise/storage.go

```diff
@@ -193,14 +199,36 @@ func (s *Storage) GetAllPromisesFromIssuer(issuerID identity.Identity) ([]Stored
 }
 
 // LoadPaymentInfo returns the last know payment information for issuer
-func (s *Storage) LoadPaymentInfo(issuerID identity.Identity) *PaymentInfo {
+func (s *Storage) LoadPaymentInfo(issuerID, consumerID identity.Identity) *PaymentInfo {
 	s.Lock()
 	defer s.Unlock()
-	sp, err := s.getLastPromise(issuerID)
+
+	defaultPaymentInfo := &PaymentInfo{
+		LastPromise: LastPromise{
+			SequenceID: 1,
+		},
+		FreeCredit: 0,
+	}
+
+	sp, err := s.findPromiseForConsumer(issuerID, consumerID)
 	if err != nil {
-		log.Trace(promiseLogPrefix, "could not load promise info, defaulting to zero payment info", err)
-		return &PaymentInfo{}
+		if err == errNoPromiseForConsumer {
+			lastPromise, lastPromiseErr := s.getLastPromise(issuerID)
+			if lastPromiseErr != nil {
+				log.Trace(promiseLogPrefix, "could not load promise info, defaulting to zero payment info ", lastPromiseErr)
+				return defaultPaymentInfo
+			}
+
+			newID := lastPromise.SequenceID + 1
+			log.Trace(promiseLogPrefix, "found payments for other consumers, will issue promiseID ", newID)
```

Comment:
Quite unreadable here, can we move sequence increment logic to storage level and lets have proper locks for that